sphinx/util/docfields.py

```diff
@@ -62,39 +65,62 @@ def __init__(self, name: str, names: Tuple[str, ...] = (), label: str = None,
 
     def make_xref(self, rolename: str, domain: str, target: str,
                   innernode: Type[TextlikeNode] = addnodes.literal_emphasis,
-                  contnode: Node = None, env: BuildEnvironment = None) -> Node:
+                  contnode: Node = None, env: BuildEnvironment = None,
+                  inliner: Inliner = None, location: Node = None) -> Node:
+        # note: for backwards compatibility env is last, but not optional
+        assert env is not None
+        assert (inliner is None) == (location is None), (inliner, location)
         if not rolename:
             return contnode or innernode(target, target)
-        refnode = addnodes.pending_xref('', refdomain=domain, refexplicit=False,
-                                        reftype=rolename, reftarget=target)
-        refnode += contnode or innernode(target, target)
-        if env:
-            env.get_domain(domain).process_field_xref(refnode)
-        return refnode
+        role = env.get_domain(domain).role(rolename)
```

Comment:
Note (for me): the `domain` is passed from DocFieldTransformer. So it surely exists. So we don't need to take care the `env.get_domain()` raises an exception.